sdk/core/core-client/test/deserializationPolicy.spec.ts

```diff
@@ -688,6 +688,64 @@ describe("deserializationPolicy", function () {
         );
       }
     });
+
+    it(`json response with headers`, async function () {
+      const BodyMapper: CompositeMapper = {
+        serializedName: "getproperties-body",
+        type: {
+          name: "Composite",
+          className: "PropertiesBody",
+          modelProperties: {
+            message: {
+              serializedName: "message",
+              type: {
+                name: "String",
+              },
+            },
+          },
+        },
+      };
+
+      const HeadersMapper: CompositeMapper = {
+        serializedName: "getproperties-headers",
+        type: {
+          name: "Composite",
+          className: "PropertiesHeaders",
+          modelProperties: {
+            foo: {
+              serializedName: "x-ms-foo",
+              type: {
+                name: "String",
+              },
+            },
+          },
+        },
+      };
+
+      const serializer = createSerializer({ HeadersMapper, BodyMapper }, false);
+
+      const operationSpec: OperationSpec = {
+        httpMethod: "GET",
+        responses: {
+          200: {
+            headersMapper: HeadersMapper,
+            bodyMapper: BodyMapper,
+          },
+        },
+        serializer,
+      };
+
+      const result = await getDeserializedResponse({
+        operationSpec,
+        headers: { "x-ms-foo": "SomeHeaderValue", "x-ms-bar": "SomeOtherHeaderValue" },
+        bodyAsText: '{"message": "Some kind of message"}',
```

Comment:
nit: do we want add an extra body property to test that it is still copied over?